src/test/java/ca/corefacility/bioinformatics/irida/service/ProjectHashingServiceIT.java

```diff
@@ -0,0 +1,116 @@
+@ActiveProfiles("it")
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class, DbUnitTestExecutionListener.class,
+        WithSecurityContextTestExecutionListener.class})
+@DatabaseSetup("/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImplIT.xml")
+@DatabaseTearDown("/ca/corefacility/bioinformatics/irida/test/integration/TableReset.xml")
+public class ProjectHashingServiceIT {
+
+    @Autowired
+    ProjectService projectService;
+    @Autowired
+    SampleService sampleService;
+    @Autowired
+    SequencingObjectService sequencingObjectService;
+
+    @Autowired
+    ProjectHashingService hashingService;
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    public void canGenerateHash() {
+        Project project = projectService.read(2L);
+        Integer expectedHash = 1654242358;
+
+        Integer projectHash = hashingService.getProjectHash(project);
+
+        assertEquals(expectedHash, projectHash);
+    }
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    public void hashChangesWithNewSample() {
+        Project project = projectService.read(2L);
+
+        Integer originalHash = hashingService.getProjectHash(project);
+
+        Sample sample = new Sample("testSample");
+        projectService.addSampleToProject(project, sample, true);
+
+        Integer newHash = hashingService.getProjectHash(project);
+
+        assertNotEquals(originalHash, newHash);
+    }
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    public void hashChangesWithRemovedSample() {
+        Project project = projectService.read(2L);
+
+        Integer originalHash = hashingService.getProjectHash(project);
+
+        List<Join<Project, Sample>> samplesForProject = sampleService.getSamplesForProject(project);
+        Sample sample = samplesForProject.iterator().next().getObject();
+
+        projectService.removeSampleFromProject(project, sample);
+
+        Integer newHash = hashingService.getProjectHash(project);
+
+        assertNotEquals(originalHash, newHash);
+    }
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    public void hashChangesWithRemovedSequencingObject() {
+        Project project = projectService.read(2L);
+
+        Integer originalHash = hashingService.getProjectHash(project);
+
+        List<Join<Project, Sample>> samplesForProject = sampleService.getSamplesForProject(project);
+        Sample sample = samplesForProject.iterator().next().getObject();
+
+        Collection<SampleSequencingObjectJoin> sequencingObjectsForSample = sequencingObjectService.getSequencingObjectsForSample(sample);
+        SequencingObject sequencingObject = sequencingObjectsForSample.iterator().next().getObject();
+
+        sampleService.removeSequencingObjectFromSample(sample, sequencingObject);
+
+        Integer newHash = hashingService.getProjectHash(project);
+
+        assertNotEquals(originalHash, newHash);
+    }
+
```

Comment:
Add a test for metadata changes for a sample as well if possible. If that part isn't good to go just leave a todo

Reply:
added in 6529954